### ISSUE TITLE
Abbreviations should have explicit case-sensitive or insensitive search option

### ISSUE THREAD

**Issue by yuryleb** [Project-OSRM/osrm-text-instructions#232]
[Documentation for new Abbreviations stuff](https://github.com/Project-OSRM/osrm-text-instructions/blob/master/languages/abbreviations/README.md#L22) has following note:

> The keys are in lowercase, to aid in case-insensitive matching

But Russian street names should be processed case-sensitively so as Russian status names are [always in lower-case](https://wiki.openstreetmap.org/wiki/RU:%D0%92%D0%B8%D0%BA%D0%B8%D0%9F%D1%80%D0%BE%D0%B5%D0%BA%D1%82_%D0%A0%D0%BE%D1%81%D1%81%D0%B8%D1%8F/%D0%A1%D0%BE%D0%B3%D0%BB%D0%B0%D1%88%D0%B5%D0%BD%D0%B8%D0%B5_%D0%BE%D0%B1_%D0%B8%D0%BC%D0%B5%D0%BD%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B8_%D0%B4%D0%BE%D1%80%D0%BE%D0%B3) to distinct them from given names. Also some 'status names' could also be 'given names' in some cases like _[Набережная улица](https://www.openstreetmap.org/way/30143450)_ where first word could mean _Embankment_ status name or just _Seafront Street_ as here.

Please also look at other ["multi-status" Russian streets collection](https://github.com/AMDmi3/streetmangler/blob/master/data/ru/multistatus.txt) prepared by [streetmangler](https://github.com/AMDmi3/streetmangler) project.

Maybe it's better to explicitly add some `meta` block to abbreviation JSONs similar to other JSON files with a `'case sensitive' : true/false` option?

**Comment by 1ec5** [comment]
Does the abbreviation customarily retain the same case as the spelled-out word? If so, we could document that the client is expected to preserve the case as it abbreviates a word. The logic would look something like this:

```js
token = token.replace(wordRegExp, function (word) {
  let abbreviatedWord = abbreviations[word];
  if (word != word.toLowerCase()) {
    abbreviatedWord = capitalizeFirstLetter(abbreviatedWord);
  }
  return abbreviatedWord;
});
```

/cc @danpaz

**Comment by yuryleb** [comment]
No, I meant _Набережная улица_ should be abbreviated as _Набережная ул._ not _Наб. ул._ :wink: But _набережная Кутузова_ should be _наб. Кутузова_.

That's it's just enough to process abbreviations as is that's case-sensitively.
